cookbooks/aws-parallelcluster-test/recipes/tests.rb

```diff
@@ -101,6 +101,42 @@
     raise "node_type must be HeadNode or ComputeFleet"
   end
 end
+###################
+# Scheduler Plugin
+###################
+if node['cluster']['scheduler'] == 'plugin'
+  if node['cluster']['node_type'] == "HeadNode"
+    execute 'check artifacts are in target_source_path' do
+      command "sudo ls #{node['cluster']['scheduler_plugin']['home']} | grep develop"
+    end
+    execute 'check handler-env.json in path' do
+      command "ls #{node['cluster']['shared_dir']}/handler-env.json"
+    end
+    execute 'check get compute fleet script in path' do
+      command "ls /usr/local/bin/get-compute-fleet-status.sh"
+    end
+    execute 'check update compute fleet script in path' do
+      command "ls /usr/local/bin/update-compute-fleet-status.sh"
+    end
+    execute "check clusterstatusmgtd is configured to be executed by supervisord" do
+      command "#{node['cluster']['cookbook_virtualenv_path']}/bin/supervisorctl start clusterstatusmgtd"
+      command "#{node['cluster']['cookbook_virtualenv_path']}/bin/supervisorctl stop clusterstatusmgtd"
+    end
+  end
+  bash "check scheduler plugin user doesn't have Sudo Privileges" do
+    cwd Chef::Config[:file_cache_path]
+    code <<-TestSudoerPrivileges
+      set -e
+      result=$((su #{node['cluster']['scheduler_plugin']['user']} -c 'sudo -ln') 2>&1 | grep 'a password is required')
+      if [ "${?}" -eq 1 ]; then
+        echo "Scheduler plugin user #{node['cluster']['scheduler_plugin']['user']} has sudo privilege"
```

Comment:
this point is not reached because with `set -e` you are exiting as soon as a command terminates with exit code != 1

Reply:
Removed set -e and changed it to one line command